src/ctap/hid/mod.rs

```diff
@@ -124,19 +101,45 @@ impl CtapHid {
 
     const CAPABILITY_WINK: u8 = 0x01;
     const CAPABILITY_CBOR: u8 = 0x04;
+    #[cfg(not(feature = "with_ctap1"))]
     const CAPABILITY_NMSG: u8 = 0x08;
+
     // Capabilitites currently supported by this device.
     #[cfg(feature = "with_ctap1")]
-    const CAPABILITIES: u8 = CtapHid::CAPABILITY_WINK | CtapHid::CAPABILITY_CBOR;
+    const CAPABILITIES: u8 = Self::CAPABILITY_WINK | Self::CAPABILITY_CBOR;
     #[cfg(not(feature = "with_ctap1"))]
-    const CAPABILITIES: u8 =
-        CtapHid::CAPABILITY_WINK | CtapHid::CAPABILITY_CBOR | CtapHid::CAPABILITY_NMSG;
+    const CAPABILITIES: u8 = Self::CAPABILITY_WINK | Self::CAPABILITY_CBOR | Self::CAPABILITY_NMSG;
 
-    // TODO: Is this timeout duration specified?
-    const TIMEOUT_DURATION: Duration<isize> = Duration::from_ms(100);
-    const WINK_TIMEOUT_DURATION: Duration<isize> = Duration::from_ms(5000);
+    const TIMEOUT_DURATION_MS: ClockInt = 100;
+    const TIMEOUT_DURATION: Milliseconds<ClockInt> = Milliseconds(Self::TIMEOUT_DURATION_MS);
+    const WINK_TIMEOUT_DURATION_MS: ClockInt = 5000;
+    const WINK_TIMEOUT_DURATION: Milliseconds<ClockInt> =
+        Milliseconds(Self::WINK_TIMEOUT_DURATION_MS);
+}
+
+pub struct CtapHid<C: Clock> {
```

Comment:
Now that we have a `clock` module exposing a `CtapClock` you don't need to abstract `CtapHid` over `Clock`. You can just directly use `CtapClock` anywhere you have `C`. This should reduce the diff considerably.

Reply:
See my comment above regarding testing.

Reply:
Done.